template/tasks.py

```diff
@@ -134,13 +134,19 @@ def check_migrations(ctx):
         "verbose": "run tests in verbose mode",
     }
 )
-def test(ctx, name="", verbose=False, suite=None):
+def test(
+    ctx,
+    name="",
+    verbose: bool = False,
+    with_deps: bool = False,
+    suite: str | None = None,
+) -> None:
     """
     Runs the test suite
     """
     if suite != "system":
         # Ensure we have a playwright browser; Skip this when running system suite only.
-        ctx.run("poetry run playwright install")
+        ctx.run(f"poetry run playwright install {' --with-deps' if with_deps else ''}")
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** In what situations do you need to pass the `--with-deps` flag?

If we are having to add parameters for the playwright installation to the `test` target I'd be inclined to remove the installation from this task entirely and create a new `install` task that does a poetry install and the playwright install.‏

Reply:
Based on [Playwright docs](https://playwright.dev/python/docs/ci-intro#on-pushpull_request) it's recommended when running tests on ci, ensure that playwright is installed along with the required dependancies.

>  create a new install task that does a poetry install and the playwright install.
Do you mean having a separate job on ci for installing them or adding a new invoke task?

Reply:
I mean a new `invoke` task. The test CI job would then be something like:

```invoke install test```

The `install` task would:

- poetry install
- playwright install --with-deps (always, no need for a parameter to the task)

The invoke `test` task would no longer do anything with playwright.

The CI job would call `invoke install` and then `invoke test`

Reply:
Added the new task: [47a54c1](https://github.com/ackama/django-template/pull/46/commits/47a54c10fc258d347a4b4135a412b2913e7393a7)